src/routes/transactions/mappers/common/native-staking.mapper.ts

```diff
@@ -306,42 +298,97 @@ export class NativeStakingMapper {
       ? StakingValidatorsExitStatus.RequestPending
       : StakingValidatorsExitStatus.ReadyToWithdraw;
   }
+
+  /**
+   * Gets the value staked in the native staking deployment based on the number of validators.
+   * Each validator has a fixed amount of 32 ETH staked.
+   *
+   * @param numValidators - the number of validators
+   * @param chain - the chain where the native staking deployment lives
+   * @returns - the value staked in the native staking deployment
+   */
+  private getValueFromNumValidators(
+    numValidators: number,
+    chain: Chain,
+  ): number {
+    const decimals = chain.nativeCurrency.decimals;
+    return (
+      numValidators *
+      Math.pow(10, decimals) *
+      NativeStakingMapper.ETH_ETHERS_PER_VALIDATOR
+    );
+  }
+
   /**
-   * Gets the total value to withdraw from the native staking deployment
-   * based on the amount of publicKey items in the transaction data.
+   * Gets the rewards to withdraw from the native staking deployment
+   * based on the length of the publicKeys field in the transaction data.
    *
-   * Each publicKey item represents a validator to withdraw, and each
-   * native staking validator has a fixed amount of 32 ETH to withdraw.
+   * Each {@link STAKING_PUBLIC_KEY_LENGTH} characters represent a validator to withdraw,
+   * and each native staking validator has a fixed amount of 32 ETH to withdraw.
    *
    * @param data - the decoded data of the transaction
    * @param chain - the chain where the native staking deployment lives
-   * @returns the total value to withdraw from the native staking deployment
+   * @returns the rewards to withdraw from the native staking deployment
    */
-  private getValueFromDataDecoded(data: DataDecoded, chain: Chain): number {
-    if (data.method === 'requestValidatorsExit') {
-      const publicKeys = this.getPublicKeysFromDataDecoded(data);
-      const decimals = chain.nativeCurrency.decimals;
-      return (
-        publicKeys.length *
-        Math.pow(10, decimals) *
-        NativeStakingMapper.ETH_ETHERS_PER_VALIDATOR
-      );
+  private async getRewardsFromDataDecoded(
+    data: DataDecoded,
+    chain: Chain,
+  ): Promise<number> {
+    const publicKeys = this.getPublicKeysFromDataDecoded(data);
+    if (!publicKeys) {
+      return 0;
     }
-
-    return 0;
+    const stakes = await this.stakingRepository.getStakes({
+      chainId: chain.chainId,
+      validatorsPublicKeys: this.splitPublicKeys(publicKeys).join(','),
```

Comment:
I totally agree these functions should be moved to the `KilnApi` or another Kiln-related class. But we would need to access that logic through the `StakingRepository` somehow, so it would require a refactor of multiple classes. I've left a `TODO` to do so in a separate PR if you agree, so we can unblock the main functionality first, wdyt?

Reply:
Sounds good to me!

Reply:
Sounds good to me. I think a lot of the adjustments here regarding public key concatenation may also need reversion though but that can be included in the subsequent PR.

In general, I understand the reasoning for extracting the relative data from `DataDecoded` but we need to be careful to remember the fallback. I think we should move that with this function.

Reply:
Sure, my main concern is most of the time the decoder doesn't actually decode the transaction data, the Transaction Service does. So we would need to centralize all the decoding/public key parsing logic in a single place/layer, but we can align for that in the forthcoming PR.

Thanks for the detailed review and comments!